src/components/ProjectVersion/index.jsx

```diff
@@ -1,19 +1,178 @@
-import React from "react";
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
 import styled from "styled-components";
 
+import BottomNavigator from "../shared/BottomNavigator";
 import Title from "../shared/Title";
 import Select from "../shared/Select";
-import BottomNavigator from "../shared/BottomNavigator";
+import Modal from "../shared/Modal";
+import Loading from "../shared/Loading";
+import { getPages } from "../../services/pages";
+import convertLabel from "../utils/convertLabel";
+
+import useProjectVersionStore from "../../../store/projectVersion";
+import usePageListStore from "../../../store/projectPage";
+import usePageStatusStore from "../../../store/projectInit";
 
 function ProjectVersion() {
+  const [isLoaded, setIsLoaded] = useState(true);
+  const [beforeDate, setBeforeDate] = useState("");
+  const [afterDate, setAfterDate] = useState("");
+  const [beforeVersion, setBeforeVersion] = useState("");
+  const [afterVersion, setAfterVersion] = useState("");
+  const navigate = useNavigate();
+
+  const { allDates, byDates } = useProjectVersionStore(state => state);
+
+  const onChangeBeforeDate = ev => {
+    const date = ev.target.value;
+
+    setBeforeDate(date);
+  };
+
+  const onChangeBeforeVersion = ev => {
+    const version = ev.target.value;
+
+    setBeforeVersion(version);
+  };
+
+  const onChangeAfterDate = ev => {
+    const date = ev.target.value;
+
+    setAfterDate(date);
+  };
+
+  const onChangeAfterVersion = ev => {
+    const version = ev.target.value;
+
+    setAfterVersion(version);
+  };
+
+  const dateFilter = () => {
+    if (beforeDate) {
+      return allDates.filter(
+        dateString => new Date(dateString) > new Date(beforeDate),
+      );
+    }
+
+    return false;
+  };
+
+  const updateVersions = (prevVersion, nextVersion) => {
+    usePageStatusStore.getState().setStatus({
+      beforeVersion: prevVersion,
+      afterVersion: nextVersion,
```

Comment:
P3: 이름을 통일하면 좋을 것 같습니다 👍
```suggestion
  const updateVersions = (beforeVersion, afterVersion) => {
    usePageStatusStore.getState().setStatus({
      beforeVersion,
      afterVersion,
```